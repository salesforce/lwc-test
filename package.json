{
    "name": "lwc-test",
    "private": true,
    "description": "LWC test plugins and utilities",
    "scripts": {
        "test": "yarn test:unit && yarn test:integration",
        "test:unit": "jest --config ./scripts/jest/root.config.js --no-cache",
        "test:integration": "jest --config ./test/jest.config.js --no-cache",
        "clean": "lerna run clean && lerna clean --yes && rm -rf node_modules",
        "lint": "eslint packages/ scripts/",
        "format": "prettier --write '{packages,scripts}/**/*.{js,ts,json,md}'",
        "release": "yarn install && ./scripts/release_npm.sh"
    },
    "workspaces": [
        "packages/@lwc/*",
        "test"
    ],
    "license": "MIT",
    "devDependencies": {
        "@lwc/compiler": "^1.10.0",
        "@lwc/engine": "^1.10.0",
        "@lwc/synthetic-shadow": "^1.10.0",
        "@lwc/wire-service": "^1.10.0",
        "@types/jest": "^26.0.20",
        "babel-eslint": "^10.1.0",
        "eslint": "^7.17.0",
        "jest": "^26.6.3",
        "lerna": "^3.22.1",
        "prettier": "^2.2.1"
    }
}
