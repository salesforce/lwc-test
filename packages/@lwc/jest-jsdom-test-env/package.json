{
    "name": "@lwc/jest-jsdom-test-env",
    "version": "18.0.1",
    "description": "Jest custom test environment for JSDOM",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-jsdom-test-env"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "src/jsdom-test-env.js",
    "peerDependencies": {
        "jest": "^28 || ^29",
        "jest-environment-jsdom": "^28 || ^29"
    },
    "engines": {
        "node": ">=10"
    }
}
