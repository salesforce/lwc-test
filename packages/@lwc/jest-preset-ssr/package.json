{
    "name": "@lwc/jest-preset-ssr",
    "version": "11.4.0",
    "description": "Jest preset configuration and stubs to help test SSR rendering on LWC",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-preset-ssr"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "jest-preset.js",
    "keywords": [
        "jest",
        "lwc",
        "ssr",
        "server"
    ],
    "files": [
        "/src/*.js"
    ],
    "peerDependencies": {
        "@lwc/compiler": "^2",
        "@lwc/engine-server": "^2",
        "jest": "^26 || ^27 || ^28"
    },
    "dependencies": {
        "@lwc/jest-resolver": "11.4.0",
        "@lwc/jest-transformer": "11.4.0"
    },
    "engines": {
        "node": ">=10"
    }
}
