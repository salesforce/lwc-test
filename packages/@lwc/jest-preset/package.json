{
    "name": "@lwc/jest-preset",
    "version": "18.0.0",
    "description": "Jest preset configuration and stubs to help test LWC",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-preset"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "jest-preset.js",
    "types": "types/index.d.ts",
    "keywords": [
        "jest",
        "lwc"
    ],
    "files": [
        "/ssr/*.js",
        "/src/**/*.js",
        "/ssr-for-hydration/**/*.js",
        "/ssr-server/**/*.js",
        "/types"
    ],
    "peerDependencies": {
        "@lwc/compiler": ">=5.0.0",
        "@lwc/engine-dom": ">=5.0.0",
        "@lwc/engine-server": ">=5.0.0",
        "@lwc/synthetic-shadow": ">=5.0.0",
        "jest": "^28 || ^29"
    },
    "dependencies": {
        "@lwc/jest-jsdom-test-env": "18.0.0",
        "@lwc/jest-resolver": "18.0.0",
        "@lwc/jest-serializer": "18.0.0",
        "@lwc/jest-transformer": "18.0.0",
        "jest-serializer-html": "7.1.0"
    },
    "engines": {
        "node": ">=10"
    }
}
