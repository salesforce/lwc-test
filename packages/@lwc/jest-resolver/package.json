{
    "name": "@lwc/jest-resolver",
    "version": "18.0.1",
    "description": "Jest resolver to resolve LWC specific packages",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-resolver"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "src/index.js",
    "keywords": [
        "jest",
        "lwc"
    ],
    "files": [
        "/resources/",
        "/src/index.js"
    ],
    "dependencies": {
        "@lwc/jest-shared": "18.0.1"
    },
    "peerDependencies": {
        "jest": "^28 || ^29"
    },
    "engines": {
        "node": ">=10"
    }
}
