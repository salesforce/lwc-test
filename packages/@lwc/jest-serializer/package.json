{
    "name": "@lwc/jest-serializer",
    "version": "18.0.0",
    "description": "Custom Jest snapshot serializer",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-serializer"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "src/index.js",
    "keywords": [
        "jest",
        "lwc"
    ],
    "dependencies": {
        "@lwc/jest-shared": "18.0.0",
        "pretty-format": "^29.7.0"
    },
    "peerDependencies": {
        "jest": "^28 || ^29"
    },
    "engines": {
        "node": ">=10"
    }
}
