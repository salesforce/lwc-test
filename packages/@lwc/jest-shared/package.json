{
    "name": "@lwc/jest-shared",
    "version": "18.0.1",
    "description": "Shared internal code for LWC Jest packages",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-shared"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "src/index.js",
    "keywords": [
        "jest",
        "lwc"
    ],
    "engines": {
        "node": ">=10"
    }
}
