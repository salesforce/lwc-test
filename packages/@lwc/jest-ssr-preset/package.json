{
    "name": "@lwc/jest-ssr-preset",
    "version": "11.4.0",
    "description": "Jest preset configuration and stubs to help test SSR rendering on LWC",
    "homepage": "https://lwc.dev/",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/lwc-test.git",
        "directory": "packages/@lwc/jest-ssr-preset"
    },
    "bugs": {
        "url": "https://github.com/salesforce/lwc-test/issues"
    },
    "license": "MIT",
    "main": "jest-preset.js",
    "keywords": [
        "jest",
        "lwc",
        "ssr",
        "server"
    ],
    "files": [
        "/src/*.js"
    ],
    "peerDependencies": {
        "@lwc/compiler": "*",
        "@lwc/engine-server": "*",
        "jest": "^26 || ^27 || ^28"
    },
    "dependencies": {
        "@lwc/jest-resolver": "11.4.0",
        "@lwc/jest-transformer": "11.4.0"
    },
    "engines": {
        "node": ">=10"
    }
}
