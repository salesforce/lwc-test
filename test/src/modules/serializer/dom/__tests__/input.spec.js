/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import semver from 'semver';
import { version as jestVersion } from 'jest/package.json';

const jestMajorVersion = semver.major(jestVersion);

describe('should return the expected state', () => {
    it('undefined', () => {
        const input = undefined;
        expect(input).toMatchInlineSnapshot('undefined');
    });
    it('null', () => {
        const input = null;
        expect(input).toMatchInlineSnapshot(`null`);
    });
    it('empty object', () => {
        const input = {};
        if (jestMajorVersion >= 29) {
            // Jest changed its snapshot format in v29
            expect(input).toMatchInlineSnapshot(`{}`);
        } else {
            expect(input).toMatchInlineSnapshot(`Object {}`);
        }
    });
    it('without the nodeType property', () => {
        const input = { anotherKindOfProperty: 5 };
        if (jestMajorVersion >= 29) {
            // Jest changed its snapshot format in v29
            expect(input).toMatchInlineSnapshot(`
                            {
                              "anotherKindOfProperty": 5,
                            }
                    `);
        } else {
            expect(input).toMatchInlineSnapshot(`
                Object {
                  "anotherKindOfProperty": 5,
                }
            `);
        }
    });
    it('with an element node type', () => {
        const input = document.createElement('div');
        expect(input).toMatchInlineSnapshot(`<div />`);
    });
    it('with a text node type', () => {
        const input = document.createTextNode('');
        expect(input).toMatchInlineSnapshot('');
    });
    it('with a comment node type', () => {
        const input = document.createComment('');
        expect(input).toMatchInlineSnapshot(`<!---->`);
    });
    it('with another kind of node type', () => {
        const input = document.createAttribute('foo');
        expect(input).toMatchInlineSnapshot(`Attr {}`);
    });
});
