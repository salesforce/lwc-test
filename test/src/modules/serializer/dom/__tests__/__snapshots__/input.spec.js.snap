// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the expected state when we have empty object 1`] = `Object {}`;

exports[`should return the expected state when we have null 1`] = `null`;

exports[`should return the expected state when we have undefined 1`] = `undefined`;

exports[`should return the expected state when we have with a comment node type 1`] = `<!---->`;

exports[`should return the expected state when we have with a text node type 1`] = ``;

exports[`should return the expected state when we have with an element node type 1`] = `<div />`;

exports[`should return the expected state when we have with another kind of node type 1`] = `Attr {}`;

exports[`should return the expected state when we have without the nodeType property 1`] = `
Object {
  "anotherKindOfProperty": 5,
}
`;
